### ISSUE TITLE
input group class

### ISSUE THREAD

**Issue by kiransp** [material-components/material-components-web#655]
Dear team,
Do we have anything equivalent to http://getbootstrap.com/components/#input-groups in MDC ?
i want a responsive full width text field with a search button attached to it. 
Can someone please help me with this ?

**Comment by pndewit** [comment]
The equivalent of these `input-groups` in Material Design are `prefixes` and `suffixes` as can be seen here: https://material.io/guidelines/components/text-fields.html#input-types-figure-caption-3

I don't think anything has been created for this yet (please correct me if I'm wrong).

For creating full-width textfields, there is something already in place: https://github.com/material-components/material-components-web/tree/master/packages/mdc-textfield#full-width

In short you can use the `mdc-textfield--fullwidth` class in combination with the `mdc-textfield` class (on the root element) to make it responsive to its (relative) parent's width.
As a side note, this `mdc-textfield--fullwidth` class also works for textareas by also adding the `mdc-textfield--multiline` class to the root element.

**Comment by kiransp** [comment]
Thanks for your answer. I want to have a "SEARCH" button along with mdc-textfield--fullwidth. Any idea on how to achieve this ?

**Comment by pndewit** [comment]
Not really sure if this is what you are looking for, but it might point you in the right direction: http://jsbin.com/zuxeyixoko/edit?html,css,output

**Comment by Garbee** [comment]
> I want to have a "SEARCH" button along with mdc-textfield--fullwidth. Any idea on how to achieve this ?

This isn't specified by the Material Design specification as far as I am aware. Something to keep in mind with MCW is it is a *library of individual components* that implement the *[Material Design specification](https://material.io/guidelines/)* for design. It isn't aimed at being a full framework like Bootstrap or Foundation. The core focus is MD and anything beyond that is up to developers to implement on their own.

When asking for a new feature or how to do a certain thing with MCW it helps if you point to the place in the specification that describes or shows what it is you are looking for.  That way it is better understood how it comes from the MD spec and if it doesn't exist currently then the work can be tasked and prioritized accordingly.

**Comment by traviskaufman** [comment]
Hey @kiransp ,

As @pndewit mentioned, we do have _prefixes and suffixes_ now, which are a new addition to the Material Design text fields specification. We are currently in the process of planning out tackling these features, and will probably have issues for them by early next week at the latest. Thanks!
